contrib/signet/getcoins.py

```diff
@@ -120,22 +122,34 @@ def bitcoin_cli(rpc_command_and_params):
 if args.captcha != '': # Retrieve a captcha
     try:
         res = session.get(args.captcha)
-    except:
-        print('Unexpected error when contacting faucet:', sys.exc_info()[0])
-        exit(1)
+        res.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        raise SystemExit(f"Unexpected error when contacting faucet: {e}")
+
+    # Size limitation
+    svg = xml.etree.ElementTree.fromstring(res.content)
+    if (svg.attrib.get('width') != '150' or svg.attrib.get('height') != '50'):
+        raise SystemExit("Captcha dimentions doesn't satisify")
+
+    # SVG sanitizing
+    content = bleach.clean(res.content.decode(), tags=['svg', 'path'], attributes={'*': ['path', 'fill', 'height', 'width', 'd']}).encode()
+    # Allowing only SVG's
+    if content == b"":
+        raise SystemExit(f"No captcha found URL {args.captcha}")
 
     # Convert SVG image to PPM, and load it
     try:
-        rv = subprocess.run([args.imagemagick, '-', '-depth', '8', 'ppm:-'], input=res.content, check=True, capture_output=True)
+        rv = subprocess.run([args.imagemagick, '-', '-depth', '8', 'ppm:-'], input=content, check=True, capture_output=True)
```

Comment:
FWIW, if you want to allow only svgs you can also replace the first `'-'` with ``svg:-'`. This makes imagemagick check the format and would avoid adding two dependencies.

Reply:
yes, okay